ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/management-security.j2

```diff
@@ -129,6 +129,45 @@
 | {{ policy.name }} | > {{ min_digits }} | > {{ min_length }} | > {{ min_lower }} | > {{ min_special }} | > {{ min_upper }} | < {{ max_repetitive }} | < {{ max_sequential }} |
 {%         endfor %}
 {%     endif %}
+{%     if management_security.shared_secret_profiles is arista.avd.defined %}
+
+### Secret Profiles
+
+| Profile Name | Secret Name | Secret Password | Receive Lifetime | Transmit Lifetime |
+| ------------ | ----------- | --------------- | ---------------- | ----------------- |
+{%         for profile in management_security.shared_secret_profiles | arista.avd.natural_sort("profile_name") %}
+{%             if profile.secrets is arista.avd.defined %}
+{%                 set temp = [] %}
+{%                 for secret in profile.secrets | arista.avd.natural_sort("name") %}
+{%                     do temp.append(secret) %}
+{%                 endfor %}
+{%                 for secret in temp %}
+{%                     if temp.index(secret) == 0 and secret.lifetime.infinite is arista.avd.defined(true) %}
+| {{ profile.profile_name }} | {{ secret.name }} | {{ secret.secret | arista.avd.hide_passwords(hide_passwords) }} | infinite | infinite |
+{%                     elif temp.index(secret) == 0 and secret.receive_lifetime is arista.avd.defined and secret.transmit_lifetime is arista.avd.defined %}
```

Comment:
Please simplify all this logic with by setting some temp variables for each field, if the relevant variables are set. Then have a single printout where you can use `<tempvar> | arista.avd.default('-')` for each field to cover the cases where it is not set.

Reply:
Fixed.